File: python/0387.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        count = [0] * 26
        for c in s:
            count[ord(c) - 97] += 1
        
        for c in s:
            if count[ord(c) - 97] == 1:
                return s.index(c)
        return -1

File: 0387.c
int firstUniqChar(char* s) {
    int index[26] = { 0 };
    int ret;
    
    for (int i = 0; i < strlen(s); i++) {
        index[s[i] - 'a']++;
    }
    for (int i = 0; i < strlen(s); i++) {
        if (index[s[i] - 'a'] == 1) {
            return i;
        }
    }
    
    return -1;
}